### DISCUSSION TITLE
Inline results of code in text`

### DISCUSSION THREAD

**Discussion by 7tupel** [nextjournal/clerk#411] [Ideas]
Hi,

I've been playing around with clerk and so far I believe it is very awesome :)

Now I found that when I create a notebook which contains some explanatory text combined with code it would be great if I would be able to inline the results of some code snippet directly in the text. Right now I can do something like this
```clojure
(def foo 123)
;; ... later
;; The value of 
foo
;; describes an example.
```

Which will surely display the value I defined earlier but it will display it in a new line like
```
The value of
123
describes an example.
```

What I would like to have is a way to put the value behind `foo` directly in the text. I.e. something like this:
```clojure
(def foo 123)
;; ... later
;; The value of `!foo` describes an example.
```
which would result in the rendered output of
```
The value of 123 describes an example.
```

Maybe I've overlooked in in the docs and it already exists but if not I believe this would make a great addition and improve readability of the rendered documents.

**Comment by zampino** [comment]
Hi @7tupel, we do have some [experiments in that direction in this notebook](https://github.com/nextjournal/clerk/blob/6ee093d3b22eb278d412f2fe5946d2b34cf937b5/notebooks/viewers/in_text_eval.clj). That replaces the default viewer for markdown `monospace` nodes and evaluates its textual content instead.
